--- a/Modules/grpmodule.c
+++ b/Modules/grpmodule.c
@@ -278,18 +278,7 @@
 static PyObject *
 grp_getgrall_impl(PyObject *module)
 {
-    PyObject *d;
-    struct group *p;
-
-    d = PyList_New(0);
-    if (d == NULL)
-        return NULL;
-
-    setgrent();
-    while ((p = getgrent()) != NULL) {
-        if (grp_universal_new(p, &d) < 0)
-            goto error;
-    }
-    endgrent();
-    return d;
-
-error:
-    endgrent();
-    Py_DECREF(d);
-    return NULL;
+    PyErr_SetString(PyExc_NotImplementedError,
+                    "grp.getgrall() is not supported on Android");
+    return NULL;
 }
